pkg/proto/transactions.go

```diff
@@ -201,7 +208,7 @@ type Transaction interface {
 	// Validate checks that all transaction fields are valid.
 	// This includes ranges checks, and sanity checks specific for each transaction type:
 	// for example, negative amounts for transfers.
-	Validate(scheme Scheme) (Transaction, error)
+	Validate(params TransactionValidationParams) (Transaction, error)
```

Comment:
On my mind `TransactionValidationParams` **may** be an interface for flexibility.

Reply:
For testing, for example. For now it doesn't matter. It's only about potential